src/printer_xml.c

```diff
@@ -54,19 +54,30 @@ modlist_add(struct mlist **mlist, const struct lys_module *mod)
         LY_CHECK_ERR_RETURN(!iter, LOGMEM(mod->ctx), EXIT_FAILURE);
         iter->next = *mlist;
         iter->module = (struct lys_module *)mod;
-        *mlist = iter;
+	iter->printed = 0;
+	*mlist = iter;
     }
 
     return EXIT_SUCCESS;
 }
 
 static void
-xml_print_ns(struct lyout *out, const struct lyd_node *node, int options)
+free_mlist(struct mlist **mlist) {
+  struct mlist *miter;
+  while(*mlist) {
+    miter = *mlist;
+    *mlist = miter->next;
+    free(miter);
+  }
```

Comment:
You'd better indent your code with 4 spaces.

Reply:
Please read the libyang coding style in [Contributing.md](https://github.com/CESNET/libyang/blob/master/CONTRIBUTING.md)